Manito/Manito/Screens/Detail-Ing/DetailngCodebaseViewController.swift

```diff
@@ -115,36 +126,43 @@ final class DetailingCodebaseViewController: BaseViewController {
         label.font = .font(.regular, ofSize: 15)
         return label
     }()
-    private let letterBoxButton: UIButton = {
+    private lazy var letterBoxButton: UIButton = {
         let button = UIButton(type: .system)
-        let action = UIAction { _ in
-            print("쪽지함!!")
+        let action = UIAction { [weak self] _ in
+            guard let roomType = self?.roomType,
+                  let roomId = self?.roomId,
+                  let mission = self?.missionContentsLabel.text
+            else {return}
+            let letterViewController = LetterViewController(roomState: roomType.rawValue, roomId: roomId, mission: mission)
+            self?.navigationController?.pushViewController(letterViewController, animated: true)
         }
         button.addAction(action, for: .touchUpInside)
-        button.setTitle("쪽지함", for: .normal)
+        button.setTitle(TextLiteral.letterViewControllerTitle, for: .normal)
         button.setTitleColor(.white, for: .normal)
         button.titleLabel?.font = .font(.regular, ofSize: 15)
         button.backgroundColor = .darkGrey002
         button.makeBorderLayer(color: .white)
         return button
     }()
-    private let manitoMemoryButton: UIButton = {
+    private lazy var manittoMemoryButton: UIButton = {
         let button = UIButton(type: .system)
-        let action = UIAction { _ in
-            print("함께 했던 기록!!")
+        let action = UIAction { [weak self] _ in
+            guard let roomId = self?.roomId else {return}
```

Comment:
p4;

```suggestion
            guard let roomId = self?.roomId else { return }
```

이거 못 참는데...

Reply:
> p4;
> 
> 이거 못 참는데...

cd2ea49 그쵸그쵸!